staging/src/kubevirt.io/api/core/v1/schema.go

```diff
@@ -1318,7 +1318,8 @@ type InterfaceBindingMethod struct {
 	Masquerade *InterfaceMasquerade `json:"masquerade,omitempty"`
 	SRIOV      *InterfaceSRIOV      `json:"sriov,omitempty"`
 	Macvtap    *InterfaceMacvtap    `json:"macvtap,omitempty"`
-	Passt      *InterfacePasst      `json:"passt,omitempty"`
+	// Deprecated, please refer to Kubevirt user guide for alternatives.
+	Passt *InterfacePasst `json:"passt,omitempty"`
```

Comment:
I think you need to add `++ optional`.
Example: https://github.com/kubevirt/kubevirt/blob/397efb8b5118b165dac9177b9a79b843a353a02f/staging/src/kubevirt.io/api/core/v1/types.go#L2663-L2665

Reply:
It should have been added from the start[1].
Better now then never.

[1] https://kubernetes.slack.com/archives/CAR30FCJZ/p1645558053184979?thread_ts=1645546692.768789&cid=CAR30FCJZ

Reply:
Not necessary, specifying `omitempty` should be enough, see this comment https://book.kubebuilder.io/reference/markers#difference-between--optional-and--kubebuildervalidationoptional

Reply:
Necessary because the convention assures other API generators will work, not only a specific one.
At least, this is my understanding from the discussion on the ref thread.

But I encourage you re-check this again in the channel.

In any case, this is not a blocker, but we should do out best to check it out.

Reply:
> Not necessary, specifying omitempty should be enough, see this comment https://book.kubebuilder.io/reference/markers#difference-between--optional-and--kubebuildervalidationoptional

Based on the kube-builder document referenced:


> If you’re using controller-gen only then they’re redundant, but if you’re using other generators or you want developers that need to build their own clients for your API, you’ll want to also include +optional.


I think we are generating our own client, so we will need the `+optional` along with omit-empty?

Reply:
> I think we are generating our own client, so we will need the `+optional` along with omit-empty?

The project is indeed building the client itself, but there may be cases where the API is used to generate other clients. While we do not have to support it, the "cost" is negligible.